go-apps/meep-rnis/sbi/rnis-sbi.go

```diff
@@ -269,14 +305,28 @@ func processActiveScenarioUpdate() {
 						cellId = ""
 					}
 
-					node := sbi.activeModel.GetNodeChild(name)
+					node := sbi.activeModel.GetNode(name)
+					ue := node.(*dataModel.PhysicalLocation)
+
+					node = sbi.activeModel.GetNodeChild(name)
 					apps := node.(*[]dataModel.Process)
 
 					var appNames []string
 					for _, process := range *apps {
 						appNames = append(appNames, process.Name)
 					}
-					sbi.updateUeDataCB(name, mnc, mcc, cellId, nrcellId, erabIdValid, appNames)
+					latency := int32(0)
+					ploss := float64(0.0)
+					throughputDL := int32(0)
+					throughputUL := int32(0)
+					if ue.NetChar != nil {
+						latency = ue.NetChar.Latency
+						ploss = ue.NetChar.PacketLoss
+						throughputDL = ue.NetChar.ThroughputDl
+						throughputUL = ue.NetChar.ThroughputUl
+					}
+
+					sbi.updateUeDataCB(UeDataSbi{name, mnc, mcc, cellId, nrcellId, erabIdValid, appNames, latency, throughputUL, throughputDL, ploss})
```

Comment:
The purpose of adding the structure was to avoid argument ordering issues. By initializing the structure this way you will continue to have ordering issues. I recommend creating an instance and populating the fields by name directly above instead of creating temporary variables.

Reply:
NOTE: Same comment for the other structures.

Reply:
fixed on all 3 structures